src/features/tokens/adapters/EvmTokenAdapter.ts

```diff
@@ -0,0 +1,180 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+import { BigNumber, PopulatedTransaction, Signer, providers } from 'ethers';
+
+import type { ERC20Upgradeable, HypERC20 } from '@hyperlane-xyz/hyperlane-token';
+import { utils } from '@hyperlane-xyz/utils';
+
+import {
+  addressToByteHexString,
+  isValidEvmAddress,
+  normalizeEvmAddress,
+} from '../../../utils/addresses';
+import {
+  getErc20Contract,
+  getHypErc20CollateralContract,
+  getHypErc20Contract,
+} from '../contracts/evmContracts';
+
+import {
+  IHypTokenAdapter,
+  ITokenAdapter,
+  TransferParams,
+  TransferRemoteParams,
+} from './ITokenAdapter';
+
+type SignerOrProvider = Signer | providers.Provider;
+
+// Interacts with native currencies
+export class EvmNativeTokenAdapter implements ITokenAdapter {
+  constructor(public readonly signerOrProvider: SignerOrProvider) {}
+
+  async getBalance(address?: Address): Promise<string> {
```

Comment:
We normalize everything to strings because diff ecosystems use diff BigNumber libs. It's better to have the adapter return strings which is 1) safe and 2) primitive and therefore won't trigger excessive re-renders in the UI